apps/api/src/secret/service/secret.service.ts

```diff
@@ -503,7 +503,59 @@ export class SecretService {
       }
     }
 
-    return secrets
+    // Group secrets by environment
```

Comment:
I feel this function can be refactored a lot. 

- Lines 494 to 503 run a for loop on the secrets and decrypt the values conditionally. I feel you can remove the top level if at line 493 and decrypt optionally in the for loop. 
- You are using another function to fetch the environment name by ID. This too isn't needed since we are already including the secret name and id in lines 479 to 484. 
- For the clustering operation, you can just employ the one that you use in variable service.

A suggestion in the naming that comes to my mind is, instead of writing `environmentId` and `environmentName`, we can create an `environment` field and include `name` and `id` as the key-value pairs.
```json
{
  "environment": {
    "name": "",
    "id": ""
  },
  ... secrets
}
```